lib/ansible/parsing/vault/__init__.py

```diff
@@ -662,7 +662,8 @@ def _edit_file_helper(self, filename, secret,
                           existing_data=None, force_save=False, vault_id=None):
 
         # Create a tempfile
-        fd, tmp_path = tempfile.mkstemp()
+        root, ext = os.path.splitext(os.path.realpath(filename))
+        fd, tmp_path = tempfile.mkstemp(suffix=ext)
```

Comment:
some people are using `.yml.vault` or similar, so the same extensions might not always apply, that said, this is probably an improvement for most people.

Reply:
@bcoca I can have it chop off `.vault` in another PR if we want to go that route. If so, do any other special extensions come to mind?